vrc/loaders/cparser.cc

```diff
@@ -35,10 +38,45 @@ void add_external_node(VisitorState *state)
                             clang_getCString(state->current_function));
 }
 
-void add_node(VisitorState *state, CXCursor c)
+std::optional<std::string> get_node_name(CXCursor c)
 {
-    size_t i = graph_add_external_node(&state->t, state->g,
-                                       clang_getCString(state->current_function));
+    std::optional<std::string> name;
+
+    CXString c_spelling = clang_getCursorSpelling(c);
+
+    switch (c.kind) {
+    case CXCursor_FunctionDecl:
+        name = clang_getCString(c_spelling);
+        break;
+
+    case CXCursor_FieldDecl:
```

Comment:
Should this check that the type is a function pointer, so that assignments are filtered nicely?